klsh/hamming_ann.py

```diff
@@ -148,7 +150,7 @@ def query(self, X, k, return_dist=False):
            an [n_samples, n_bits] array of hamming features. These will be
            interpreted as zeros and ones.
         """
-        X_compact = self._validate_input(X)
+        X_compact = self._validate_input(X, False)
```

Comment:
It would make more sense to call the variable ``X`` at this point if it's not compact.